### ISSUE TITLE
ButtonTheme.bar uses accent color when it should be using primary color

### ISSUE THREAD

**Issue by darrenaustin** [flutter/flutter#37333]
By default buttons in a `ButtonTheme.bar` theme use `ThemeData.accentColor` for their text color. This can lead to buttons that are hard to read, as they don't follow the accessibility guidelines for contrast against the background. The [Material spec](https://material.io/design/components/dialogs.html) indicates we should be using the primary color for the button text color instead.  Example of the difference using the baseline Material color scheme:

| ButtonTextTheme.accent  | ButtonTextTheme.primary |
| ------------- | ------------- |
| ![accent](https://user-images.githubusercontent.com/19588/62242104-51779f00-b38f-11e9-98b5-b7ca1c5b886c.png) | ![primary](https://user-images.githubusercontent.com/19588/62242120-563c5300-b38f-11e9-9950-ee4c333f7a7f.png) |

To fix this we need to change the default parameter for `ButtonTheme.bar` to use `ButtonTextTheme.primary` instead of `ButtonTextTheme.accent`. This would be a breaking change for existing apps that depend on `ButtonTheme.bar` defaulting to the accent color.
